luarules/gadgets/unit_collision_damage_behavior.lua

```diff
@@ -106,25 +119,40 @@ function gadget:UnitPreDamaged(unitID, unitDefID, unitTeam, damage, paralyzer, w
 end
 
 function gadget:UnitLoaded(unitID, unitDefID, unitTeam, transportID, transportTeam)
-	transportingUnits[transportID] = transportingUnits[transportID] or {}
-	transportingUnits[transportID][unitID] = true
+	transportedUnits[unitID] = true
 end
 
 function gadget:UnitUnloaded(unitID, unitDefID, unitTeam, transportID, transportTeam)
-	transportingUnits[transportID][unitID] = nil
+	transportedUnits[unitID] = nil
 end
 
 function gadget:UnitDestroyed(unitID)
-	transportingUnits[unitID] = nil
+	transportedUnits[unitID] = nil
+	unitInertiaCheckFlags[unitID] = nil
 end
 
 function gadget:GameFrame(frame)
-	if frame % 5 == 3 then
-		for unitID, data in pairs (overImpulseCooldowns) do
-			if data.expireframe < frame  then --this is done to prevent a unit from being impulsed multiple times in a compounding fashion in a performant way.
-				overImpulseCooldowns[unitID] = nil
+	for unitID, expirationFrame in pairs(unitInertiaCheckFlags) do
+		if not transportedUnits[unitID] then
+			local velocityCap = 3
+			local velX, velY, velZ, velocityLength = spGetUnitVelocity(unitID)
+			if velocityLength > velocityCap then
+				velX = (velocityCap / velocityLength) * velX
+				velY = (velocityCap / velocityLength) * velY
+				velZ = (velocityCap / velocityLength) * velZ
+			end
+			if velocityLength > objectCollisionVelocityThreshold then
+				local decelerateHorizontal = 0.95
+				local decelerateVertical = 0.33
+				if velY < 0 then
+					decelerateVertical = 1
+				end
+				spSetUnitVelocity(unitID, (velX * decelerateHorizontal), (velY * decelerateVertical), (velZ * decelerateHorizontal))
```

Comment:
```suggestion
				spSetUnitVelocity(unitID, velX * decelerateHorizontal, velY * decelerateVertical, velZ * decelerateHorizontal)
```
Parentheses are only needed to delay the processing of a higher precedence operator

Reply:
I put them there for readability. you no like?

Reply:
whatever, removed them just so I can push this along